flutter-studio/src/io/flutter/module/FlutterModuleModel.java

```diff
@@ -5,27 +5,215 @@
  */
 package io.flutter.module;
 
+import com.android.tools.idea.gradle.parser.BuildFileKey;
+import com.android.tools.idea.gradle.parser.Dependency;
+import com.android.tools.idea.gradle.parser.GradleBuildFile;
+import com.android.tools.idea.gradle.parser.GradleSettingsFile;
+import com.android.tools.idea.gradle.parser.Repository;
+import com.intellij.execution.process.OSProcessHandler;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.command.WriteCommandAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.InvalidDataException;
+import com.intellij.openapi.util.io.FileUtilRt;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import io.flutter.FlutterUtils;
+import io.flutter.actions.FlutterBuildActionGroup;
 import io.flutter.project.FlutterProjectCreator;
 import io.flutter.project.FlutterProjectModel;
+import io.flutter.pub.PubRoot;
+import io.flutter.sdk.FlutterSdk;
+import io.flutter.utils.AndroidUtils;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
 import org.jetbrains.annotations.NotNull;
 
 public class FlutterModuleModel extends FlutterProjectModel {
   public FlutterModuleModel(@NotNull FlutterProjectType type) {
     super(type);
   }
 
+  @Override
+  public boolean shouldOpenNewWindow() {
+    return !isModule();
+  }
+
   @Override
   protected void handleFinished() {
     // Do not call the superclass method.
-    if (projectType().get().isPresent() && projectType().get().get() == FlutterProjectType.IMPORT) {
-      String location = projectLocation().get();
-      assert (!location.isEmpty());
+    if (isModule()) {
+      // The FlutterProjectType.MODULE is used in two places. In FlutterProjectModel it is used to create a new top-level
```

Comment:
Is `handleFinished()` long running? Can or should anything here be done in a background task, w/ progress reported to the user?

Reply:
Yes, it runs a while. Progress is reported via the console and the status bar at the bottom of the window. It is in here that the error is generated. I'm thinking it might be better to wrap a modal progress indicator around all this, but that's another issue.